### PR TITLE
Add protocols option to bypass is-absolute package regex

### PR DESCRIPTION
**Pull request by zslabs**
Issue: https://github.com/remarkjs/remark-external-links/issues/9

Was getting some false-positives with the `indexOf` example provided; any issue with going with `includes()`? 

Thanks!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -62,52 +62,39 @@ Pass `false` to not set `rel`s on links.
...
+Protocols used to filter against the [is-absolute](https://www.npmjs.com/package/is-absolute) package.
+(`Array.<string>`, default: `['http', 'https']`).
+
 ## Contribute
 
 See [`contributing.md` in `remarkjs/remark`][contributing] for ways to get
 started.
 
-This organisation has a [Code of Conduct][coc].  By interacting with this
+This organisation has a [Code of Conduct][coc]. By interacting with this
```
Comment (wooorm) [inline readme.md]:
```suggestion
This organisation has a [Code of Conduct][coc].  By interacting with this
```

File: readme.md [hunk 2]
```diff
@@ -62,52 +62,39 @@ Pass `false` to not set `rel`s on links.
 > When using a `target`, add [`noopener` and `noreferrer`][mdn-a] to avoid
 > exploitation of the `window.opener` API.
 
+###### `options.protocols`
+
+Protocols used to filter against the [is-absolute](https://www.npmjs.com/package/is-absolute) package.
```
Comment (wooorm) [inline readme.md]: I don’t think there’s a need to mention is-absolute. That’s internals. I suggest mentioning that people can allow other protocols here, such as `mailto`, or others, instead?

File: index.js [hunk 3]
```diff
@@ -29,7 +31,11 @@ function externalLinks(options) {
       var data
       var props
 
-      if (ctx && isAbsoluteURL(ctx.url)) {
+      if (
+        ctx &&
+        isAbsoluteURL(ctx.url) &&
+        protocols.includes(ctx.url.slice(0, ctx.url.indexOf(':')))
```
Comment (wooorm) [inline index.js]: `indexOf` has better support across vendors, so I suggest using that. Maybe try a +1 or -1?
`f = 'https://example.com'; f.slice(0, f.indexOf(':'))` seems to work and return `https`, so it should work?

File: readme.md [hunk 4]
```diff
@@ -62,52 +62,39 @@ Pass `false` to not set `rel`s on links.
...
-This organisation has a [Code of Conduct][coc].  By interacting with this
+This organisation has a [Code of Conduct][coc]. By interacting with this
 repository, organisation, or community you agree to abide by its terms.
 
 ## License
 
 [MIT][license] © [Cédric Delpoux][author]
 
 [build-badge]: https://img.shields.io/travis/remarkjs/remark-external-links.svg
-
```
Comment (wooorm) [inline readme.md]: I’m guessing prettier did this? Could you turn that off (`npm run test` should format the markdown the way this project likes it)

**Review by wooorm — CHANGES_REQUESTED** [review]

**Comment by zslabs** [comment]
Thanks for the review! I've pushed an update that pulls out the protocol into its own variable; for some reason it's getting confused when thrown in with the `indexOf`.

File: readme.md (hunk 4 above)
Comment (zslabs) [reply readme.md]: Sure thing, pushed in my latest commit.

File: readme.md (hunk 2 above)
Comment (zslabs) [reply readme.md]: Sure thing, pushed in my latest commit.

**Review by wooorm — APPROVED** [review]

**Pull request merged**